File: backend/src/main/java/isis/projet/backend/entity/Utilisateur.java
package isis.projet.backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import java.util.List;

@Entity
@Data
public class Utilisateur {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idPersonne;

    private String pseudo;
    private String email;
    private String mdp;
    private String preferences;

    @OneToMany(mappedBy = "utilisateur", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Session> sessions;

    @OneToMany(mappedBy = "utilisateur", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<IndicateurGlobal> indicateursGlobaux;

    @OneToMany(mappedBy = "utilisateur", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<IndicateurSession> indicateursSession;
}

File: frontend/src/Utilisateur.js
export class Utilisateur {
  constructor(idPersonne, pseudo, email, mdp, preferences) {
    this._idPersonne = idPersonne;
    this._pseudo = pseudo;
    this._email = email;
    this._mdp = mdp;
    this._preferences = preferences;
  }

  get idPersonne() {
    return this._idPersonne;
  }

  get pseudo() {
    return this._pseudo;
  }

  get email() {
    return this._email;
  }

  get mdp() {
    return this._mdp;
  }

  get preferences() {
    return this._preferences;
  }

  toString() {
    return `--> ${this._pseudo} `;
  }
}

export default Utilisateur;
